src/server/modules/sendResponse.ts

```diff
@@ -184,6 +189,24 @@ export class SendResponse extends EventEmitter {
     return ackMessage;
   }
 
+  /** @internal */
+  private _validateMSA1(spec: string, type: Valid_MSA_1): void {
+    switch (spec) {
+      case "2.1":
+        if (!MSA_1_VALUES_v2_1.includes(type)) {
+          throw new Error(`Invalid MSA-1 value: ${type} for HL7 version 2.1`);
```

Comment:
The error should throw an HL7ServerError not a standard error. Update this can create a unit test that checks to make sure this error is thrown.

Reply:
Sure, I'll update it 👏  do you have any example on how to check whether a specific error is thrown? I'm not used to work with `vitest`

Reply:
I think we can't test this behaviour with the current implementation. The error is catched by `sendResponse` so we can't check whether this method throws an error.

The alternative would be extending the catch clause, checking if the error type is HL7Server, and if so throw it again. The execution fails and we can track the error, which in fact makes the code more predictive IMO than just sending an ACK with errors.

Wdyt?

Reply:
the server shouldn't be sending it out a valid message to the end user (client) so it will throw before hand. I am thinking now _maybe_ listener error so that the client knows something was wrong and not break the server. I'll have to test it this weekend. I have an app that does about million different HL7 messages and I will put these changes through it